test/sched.py

```diff
@@ -105,3 +117,33 @@ def test_get_node_list(self):
         inst = getinstance(mpim.MPI, Local, MympirunOption())
         self.assertEqual(set(inst.nodes), set(['localhost']))
         self.assertEqual(len(inst.cpus), len(inst.nodes))
+
+    def test_set_node_list(self):
+        """
+        test different scenarios for setting node list
+        """
+        nodes = [
+            'node1',
+            'node1',
+            'node2',
+            'node3',
+        ]
+        pbs_class = SCHEDDICT['pbs']
+        text = '\n'.join(nodes)
+        nodefile = open(os.environ['PBS_NODEFILE'], 'w')
+        nodefile.seek(0)
+        nodefile.write(text)
+        nodefile.close()
+
+        inst = getinstance(mpim.MPI, pbs_class, MympirunOption())
+        inst.set_mpinodes()
+        self.assertEqual(inst.mpinodes, nodes)
+
+        inst.options.double = True
+        inst.set_mpinodes()
+        self.assertEqual(inst.mpinodes, nodes + nodes)
```

Comment:
add a clear comment above each block to clarify which aspect your verifying exactly